awscli/autocomplete/parser.py

```diff
@@ -126,8 +126,9 @@ class CLIParser(object):
     not a general purpose AWS CLI parser.
 
     """
-    def __init__(self, index):
+    def __init__(self, index, first_wins=False):
```

Comment:
I feel like we can make a better name here. Mainly it is a little ambiguous about what is first and how does it win. How about `return_first_command_match`? It's a little lengthy but I think it represents what we are going for. I also think it might be worth adding docstrings to describe the parameter because it is a pretty specific parameter.